### PR TITLE
fix: creating multiple users on atlas connect [MCP-280]

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

sometimes atlas connect tool would create multiple users instead of one, this fixes it.

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)

### CODE REVIEW

File: src/tools/atlas/connect/connectCluster.ts [hunk 1]
```diff
@@ -218,6 +218,39 @@ export class ConnectClusterTool extends AtlasToolBase {
...
+            case "disconnected": {
+                await this.session.disconnect();
+
+                const { connectionString, atlas } = await this.prepareClusterConnection(
+                    projectId,
+                    clusterName,
+                    connectionType
+                );
+
+                createdUser = true;
```
[the remark this answers is not in the document]
Comment (fmenezes) [reply src/tools/atlas/connect/connectCluster.ts:233]: prepareClusterConnection will always create a user

File: src/tools/atlas/connect/connectCluster.ts [hunk 2]
```diff
@@ -218,6 +218,39 @@ export class ConnectClusterTool extends AtlasToolBase {
         const ipAccessListUpdated = await ensureCurrentIpInAccessList(this.session.apiClient, projectId);
         let createdUser = false;
 
+        const state = this.queryConnection(projectId, clusterName);
```
Comment (blva) [inline src/tools/atlas/connect/connectCluster.ts:221]: could we also add integration test coverage to avoid regressions in the future? 

We could improve coverage with:
 1. validate that after we connect, only on DB user is created 
 2. have a scenario tun where we connect to a cluster then disconnect and validate that the DB user is deleted

Comment (fmenezes) [reply src/tools/atlas/connect/connectCluster.ts:221]: done!

File: tests/integration/tools/atlas/clusters.test.ts [hunk 3]
```diff
@@ -172,6 +187,58 @@ describeWithAtlas("clusters", (integration) => {
...
             });
 
+            describe("when connected", () => {
+                withCluster(
+                    integration,
+                    ({ getProjectId: getSecondaryProjectId, getClusterName: getSecondaryClusterName }) => {
+                        beforeAll(async () => {
+                            let connected = false;
+                            for (let i = 0; i < 10; i++) {
+                                const response = await integration.mcpClient().callTool({
```
Comment (blva) [inline tests/integration/tools/atlas/clusters.test.ts:197]: [q] why do we need to call connect several times, shouldn't we connect and wait a considerable amount of time and that should be enough?

File: tests/integration/tools/atlas/clusters.test.ts [hunk 4]
```diff
@@ -172,6 +187,58 @@ describeWithAtlas("clusters", (integration) => {
...
+
+                                await sleep(500);
+                            }
+
+                            if (!connected) {
+                                throw new Error("Could not connect to cluster before tests");
+                            }
+                        });
+
+                        it("disconnects before connecting to another cluster", async () => {
```
Comment (blva) [inline tests/integration/tools/atlas/clusters.test.ts]:
```suggestion
                        it("disconnects and deletes the database user before connecting to another cluster", async () => {
```

**Review by blva — APPROVED** [review]
non-blocking comments, thanks for the fix and coverage!

File: tests/integration/tools/atlas/clusters.test.ts (hunk 3 above)
Comment (fmenezes) [reply tests/integration/tools/atlas/clusters.test.ts:197]: this is the edge case where atlas can take a long time to provision the DB user to data plane, in which case we need to keep waiting
